data/ai/micro_ais/cas/ca_hunter.lua

```diff
@@ -60,8 +60,7 @@ function ca_hunter:execution(cfg)
     -- If hunting_status is not set for the hunter -> default behavior -> random wander
     if (not hunter_vars.hunting_status) then
         -- Hunter gets a new goal if none exist or on any move with 10% random chance
-        local rand = math.random(10)
-        if (not hunter_vars.goal_x) or (rand == 1) then
+        if (not hunter_vars.goal_x) or (math.random(10) == 1) then
```

Comment:
Lua uses short-circuit evaluation [confirmation](http://www.lua.org/manual/5.4/manual.html#3.4.5). Could the reordering and short-circuit here cause OOS in replays?

Reply:
The first condition would either be evaluated to true or false the same in the initial play-through vs the replay, so it shouldn't?

Reply:
You're right, I was confused. Couldn't happen on a stable branch.

Faulty logic: while the WML would be loaded from the replay, the Lua code is probably read from the game data instead. But that's broken logic, as long as this is only on a development branch when getting a new version is expected to OOS some replays.